assets/src/components/ui/toggle-button/toggle-button.js

```diff
@@ -0,0 +1,40 @@
+/**
+ * ToggleButton
+ * Returns a component that can be used to trigger an action like opening or closing a container.
+ * The component appears as an icon only but acts like a button.
+ *
+ * @function
+ * @param {string} htmlId               Used to generate an HTML "id" attribute for all components.
+ *                                      Additional text is appended by each component.
+ * @param {string} htmlClass            Used to generate an HTML "class" attribute for all components.
+ *                                      Additional text is appended by each component.
+ * @param {string} description          Two or three words (hyphenated) that describe the container
+ *                                      that the ToggleButton is associated with.
+ * @param {Function} toggleCallback 	Function called when the "button" component is triggered
+ * @param {string} dashicon             Css class that defines the dashicon icon
+ * @param {Number} tabIndex             tab index for the element that controls when it receives focus
+ * @return {Object}                     "button" for opening the modal container
+ */
+export const ToggleButton = (
+	{
+		htmlId,
+		htmlClass,
+		description,
+		toggleCallback,
+		dashicon = 'no',
+		tabIndex = 0,
+	}
+) => {
+	return toggleCallback && (
```

Comment:
Should verify `toggleCallback` is a function and if it isn't use `warning` to inform maybe and convert the incoming prop to use `noop` so that there is not javascript error?

Reply:
good idea, will do.

have you used `warning` yet anywhere? I had tried to use it before but something was not set up correctly.

What specifically are you referring to by `noop` ?
I'm assuming you mean `react-props-noop` but I don't see that currently listed in our dependencies anywhere. 
Or did you mean `jQuery.noop()` ?
Or something else.

Reply:
> have you used warning yet anywhere? I had tried to use it before but something was not set up correctly.

- `assets/src/components/form/select/build-options.js)

Also the "coming soon" money and currency vo's I'm working on use warning (I'm still writing tests for them).

> What specifically are you referring to by noop ?

`lodash.noop`: https://lodash.com/docs/4.17.10#noop